File: Ombudsman_Agency_Project2.java
package ouvidoriafacisa;

import java.util.ArrayList;

public class OuvidoriaUnifacisa {
	public static void main(String[] args) {

		ArrayList<Teste> listaManifestacoes = new ArrayList<>();
		ArrayList<Teste> listaSugestao = new ArrayList<>();
		ArrayList<Teste> listaReclamacao = new ArrayList<>();
		ArrayList<Teste> listaElogio = new ArrayList<>();

		while (true) {
			System.out.println("Bem vindo a Ouvidoria Unifacisa!\n");
			System.out.println("1) Criar Manifestação");
			System.out.println("2) Listar Manifestações");
			System.out.println("3) Listar Sugestões");
			System.out.println("4) Listar Reclamações");
			System.out.println("5) Listar Elogios");
			System.out.println("6) Pesquisar Manifestação pelo ID");
			System.out.println("7) Sair\n");
			System.out.println("Digite sua opção: ");
			int opcao = Teclado.leInt();

			if (opcao == 1) {
				while (true) {
					System.out.println("Escolha o tipo de Manifestação: ");
					System.out.println("1) Sugestão");
					System.out.println("2) Reclamação");
					System.out.println("3) Elogio");
					int tipo = Teclado.leInt();

					if (tipo == 1) {
						Teste novaSugestao = new Teste();
						System.out.println("Digite seu nome: ");
						novaSugestao.nome = Teclado.leString();
						System.out.println("Descreva sua Sugestão: ");
						novaSugestao.descricao = Teclado.leString();
						System.out.println("Sugestao registrada com sucesso!\n");
						listaSugestao.add(novaSugestao);
						listaManifestacoes.add(novaSugestao);
						break;

					} else if (tipo == 2) {
						Teste novaReclamacao = new Teste();
						System.out.println("Descreva sua Reclamação: ");
						novaReclamacao.descricao = Teclado.leString();
						System.out.println("Reclamação registrada com sucesso!\n");
						listaReclamacao.add(novaReclamacao);
						listaManifestacoes.add(novaReclamacao);
						break;

					} else if (tipo == 3) {
						Teste novoElogio = new Teste();
						System.out.println("Descreva sua Elogio: ");
						novoElogio.descricao = Teclado.leString();
						System.out.println("Elogio registrado com sucesso!\n");
						listaReclamacao.add(novoElogio);
						listaManifestacoes.add(novoElogio);
						break;

					} else {
						System.out.println("Opcão inválida, tenta novamente :D");
					}
				}
			}
			if (opcao == 2) {
				int indice = 0;
				System.out.println("Manifestações Existentes:");
				for (Teste m : listaManifestacoes) {
					indice++;
					System.out.println(
							"ID: " + indice + " - " + "Requerente: " + m.nome + " - Descrição: " + m.descricao);
				}
			} else if (opcao == 3) {
				int indice = 0;
				System.out.println("Sugestões realizadas: ");
				for (Teste s : listaSugestao) {
					indice++;
					System.out.println("ID: " + indice + " - Descrição: " + s.descricao);
				}
			} else if (opcao == 4) {
				int indice = 0;
				System.out.println("Reclamações realizadas: ");
				for (Teste r : listaReclamacao) {
					indice++;
					System.out.println("ID: " + indice + " - Descrição: " + r.descricao);
				}
			} else if (opcao == 5) {
				int indice = 0;
				System.out.println("Elogios realizados: ");
				for (Teste e : listaElogio) {
					indice++;
					System.out.println("ID: " + indice + " - Descrição: " + e.descricao);
				}
			} else if (opcao == 6) {
				System.out.println("Digite o ID da manifestação: ");
				int indice = Teclado.leInt() - 1;
				Teste i = listaManifestacoes.get(indice);
				System.out.println("Requerente: " + i.nome + " - " + i.descricao);

			} else if (opcao == 7) {
				System.out.println("Fim do programa!");
				break;
			} else {
			}
		}
	}
}

File: Ombudsman_Agency_Project2.py
from opearacoesbd import *
conn = abrirBancoDados('localhost','root','qualquercoisa','ouvidoria')

opcao = 0
tipos = ['Sugestão', 'Reclamação', 'Elogio']

while opcao != 7:

    print('Bem vindo a Ouvidoria Unifacisa')
    print()
    print('1) Listar todas as manifestações')
    print('2) Listar todas as sugestões')
    print('3) Listar todas as reclamações')
    print('4) Listar todos os elogios')
    print('5) Enviar uma manifestação')
    print('6) Pesquisar protocolo por número ID')
    print('7) Sair')

    print()
    opcao = int(input('Qual opção você deseja continuar: '))
    print()

    if opcao < 1 or opcao > 7:
        print('Entrada inválida. Favor digitar uma das opções abaixo!')
        print()

    if opcao == 1:
        print('Todas as manifestações encontradas:')
        print()
        sql = "SELECT * FROM ouvidoria"
        resultado = listarBancoDados(conn, sql)

        for elemento in resultado:
            print('ID: {0}, Nome: {1}, Tipo: {2}, Descricao: {3}'.format(*elemento))

    elif opcao == 2:
        sql = "SELECT * FROM ouvidoria WHERE tipo = 'Sugestão'"
        resultado = listarBancoDados(conn, sql)

        for elemento in resultado:
            print('ID: {0}, Nome: {1}, Tipo: {2}, Descricao: {3}'.format(*elemento))

    elif opcao == 3:
        sql = "SELECT * FROM ouvidoria WHERE tipo = 'Reclamação'"
        resultado = listarBancoDados(conn, sql)

        for elemento in resultado:
            print('ID: {0}, Nome: {1}, Tipo: {2}, Descricao: {3}'.format(*elemento))

    elif opcao == 4:
        sql = "SELECT * FROM ouvidoria WHERE tipo = 'Elogio'"
        resultado = listarBancoDados(conn, sql)

        for elemento in resultado:
            print('ID: {0}, Nome: {1}, Tipo: {2}, Descricao: {3}'.format(*elemento))

    elif opcao == 5:
        nome = input('Digite seu nome: ')
        print()
        print('Tipo de manifestação:')
        print()
        quantidadeTipos = len(tipos)

        for t in range(quantidadeTipos):
            print(t + 1, ')', tipos[t])
            print()

        tipoEscolhido = int(input('Digite sua opção: '))
        print()
        tipo = tipos[tipoEscolhido - 1]

        descricao = input('Digite a descrição: ')
        print()
        print('Manifesto criado com sucesso!')
        print()

        sql = "INSERT INTO ouvidoria (nome,tipo,descricao) VALUES (%s, %s, %s)"
        dados = (nome, tipo, descricao)
        insertNoBancoDados(conn, sql, dados)

    elif opcao == 6:
        pesquisa = int(input('Digite seu número de protocolo(ID): '))

        sql = "SELECT * FROM ouvidoria WHERE id = {0}".format(pesquisa)
        resultado = listarBancoDados(conn, sql)

        for elemento in resultado:
            print('ID: {0}, Nome: {1}, Tipo: {2}, Descricao: {3}'.format(*elemento))


encerrarBancoDados(conn)

print('Fim do programa!')
